chemprop/train/evaluate.py

```diff
@@ -31,68 +35,89 @@ def evaluate_predictions(preds: List[List[float]],
     """
     info = logger.info if logger is not None else print
 
-    metric_to_func = {metric: get_metric_func(metric) for metric in metrics}
+    metric_to_func = {metric_label: get_metric_func(metric_label) for metric_label in metrics}
 
     if len(preds) == 0:
-        return {metric: [float('nan')] * num_tasks for metric in metrics}
+        return {metric: [float("nan")] * num_tasks for metric in metrics}
 
-    # Filter out empty targets for most data types, excluding dataset_type spectra
-    # valid_preds and valid_targets have shape (num_tasks, data_size)
-    if dataset_type != 'spectra':
-        valid_preds = [[] for _ in range(num_tasks)]
-        valid_targets = [[] for _ in range(num_tasks)]
-        for i in range(num_tasks):
-            for j in range(len(preds)):
-                if targets[j][i] is not None:  # Skip those without targets
-                    valid_preds[i].append(preds[j][i])
-                    valid_targets[i].append(targets[j][i])
+    preds = np.array(preds, dtype=float)
+    targets = np.array(targets, dtype=float)
+    if gt_targets is not None:
+        gt_targets = np.array(gt_targets, dtype=bool)
+    if lt_targets is not None:
+        lt_targets = np.array(lt_targets, dtype=bool)
 
     # Compute metric. Spectra loss calculated for all tasks together, others calculated for tasks individually.
     results = defaultdict(list)
-    if dataset_type == 'spectra':
-        for metric, metric_func in metric_to_func.items():
-            results[metric].append(metric_func(preds, targets))
+    if dataset_type == "spectra":
+        for metric_label, metric_func in metric_to_func.items():
+            results[metric_label].append(metric_func(preds, targets))
```

Comment:
```suggestion
        results = {metric_label: metric_func(preds, targets) for metric_label, metric_func in metric_to_func.items()}
```
this new suggestion isn't right, but it seems like the original code produces a value of `results: dict[str, [float]]`. I.e., that each value was a singleton list, is that right?

Reply:
we can also return right after this block and de-indent the remaining code to give it a little more breathing room